datadog_checks_base/datadog_checks/base/stubs/aggregator.py

```diff
@@ -292,10 +292,15 @@ def _assert(condition, msg, expected_stub, submitted_elements):
         assert condition, new_msg
 
     def assert_all_metrics_covered(self):
-        missing_metrics = ''
-        if self.metrics_asserted_pct < 100.0:
-            missing_metrics = self.not_asserted()
-        assert self.metrics_asserted_pct >= 100.0, 'Missing metrics: {}'.format(missing_metrics)
+        # use `condition` to avoid building the `msg` if not needed
+        condition = self.metrics_asserted_pct >= 100.0
+        msg = ''
+        if not condition:
+            prefix = '\n\t- '
+            msg = 'Some metrics are missing:'
+            msg += '\nAsserted Metrics:{}{}'.format(prefix, prefix.join(sorted(self._asserted)))
```

Comment:
🏅 Awesome! I think it might be helpful to only print the missing metrics, since this message could get pretty long.

Reply:
I think printing asserted metrics gives you some more context to debug the issue when doing testing and on CI failure.

Usually it's not that long since we are printing a metric name, not all submitted metrics.

Reply:
👍